d20/Manual/Facts/__init__.py

```diff
@@ -44,20 +46,22 @@ def isFactGroup(arg):
     return False
 
 
-def resolveFacts(*args):
+def resolveFacts(*args) -> List[str]:
     """Function to expand and consilidate facts and fact groups
 
         This function takes a list of facts or fact groups as either an
         iterable or variadic arguments and returns a list of all facts,
         converting fact groups into individual facts it represents
     """
+    facts: Union[Tuple, Iterable]
     if (len(args) == 1 and
             not isinstance(args[0], str) and
             isinstance(args[0], Iterable)):
-        args = args[0]
-
-    resolved = []
-    for fact in args:
+        facts = args[0]
+    else:
+        facts = args
+    resolved: List[str] = []
```

Comment:
aren't the values of `RegisteredFactGroups` a `Set` and not a `str`?  so shouldn't this hint cover two possible types in the list? (which also would affect the function return hint as well), so something like this:

`resolved: Sequence[Union[Set,str]] = []`

Reply:
The values of `RegisteredFactGroups` are a `Set`, but on line 64 the `Set` is converted to a `List` before being extended into the `resolved` variable, so I think it should still be `resolved: List[str] = []`

Reply:
Apologies I must have missed the casting, or something as looking at the comment and the code either way it would take the values of the set/list.  Also since it is extending resolve, you don't need to cast to the list, and can just extend passing the set as it is an iterable type (and would be faster as well).

Reply:
old: `resolved.extend(list(RegisteredFactGroups[fact]))`

new: `resolved.extend(RegisteredFactGroups[fact])`